### PR TITLE
turn on garbage collection by default

### PR DESCRIPTION
**Pull request by joelddiaz**
If you want to disable it unset
FEATURE_ENABLE_GARBAGE_COLLECTION in the operator Deployment.

Signed-off-by: Joel Diaz <joel@mondoo.com>

### CODE REVIEW

**Review by vjeffrey — DISMISSED** [review]

File: pkg/feature_flags/feature_flags.go [hunk 1]
```diff
@@ -63,9 +67,19 @@ func setGlobalFlags(k, v string) {
...
+			enableGarbageCollection = false
+		} else {
+			enableGarbageCollection = true
+		}
 	case "FEATURE_ENABLE_ADMISSION_REVIEW_DISCOVERY":
 		enableAdmissionReviewDiscovery = true
 	}
 }
+
+func setDefaults() {
```
Comment (vjeffrey) [inline pkg/feature_flags/feature_flags.go:81]: 👍

**Review by vjeffrey — DISMISSED** [review]

File: pkg/feature_flags/feature_flags.go [hunk 2]
```diff
@@ -63,9 +67,19 @@ func setGlobalFlags(k, v string) {
 		enablePodDiscovery = true
 	case "FEATURE_DISCOVER_WORKLOADS":
 		enableWorkloadDiscovery = true
-	case "FEATURE_ENABLE_GARBAGE_COLLECTION":
-		enableGarbageCollection = true
+	case FeatureGarbageCollection:
```
Comment (imilchev) [inline pkg/feature_flags/feature_flags.go:70]: I think this change is not really needed. The way I enabled workload discovery is by just setting the env var to "1" in the deployment manifest. https://github.com/mondoohq/mondoo-operator/blob/main/config/manager/manager.yaml#L36-L37 I think that's much cleaner than altering the code

Comment (joelddiaz) [reply pkg/feature_flags/feature_flags.go:70]: I suppose that is a lot simpler than my approach. I'll update/rework the PR.

**Review by imilchev — APPROVED** [review]
Sweet! Then when we know it works all good, we just drop the flags completely!

**Pull request merged**